python/sglang/srt/models/qwen3_vl.py

```diff
@@ -331,89 +332,30 @@ def rot_pos_emb(self, grid_thw):
         return rotary_pos_emb
 
     def fast_pos_embed_interpolate(self, grid_thw):
-        num_grid_per_side = int(self.num_position_embeddings**0.5)
-
-        idx_list = [[] for _ in range(4)]
-        weight_list = [[] for _ in range(4)]
+        patch_pos_embeds_permute = []
+        m_size = self.spatial_merge_size
 
-        # TODO: use torch instand of np
+        embeds = torch.arange(self.num_grid, device=self.pos_embed.weight.device)
+        embeds = (
+            self.pos_embed(embeds)
+            .permute(1, 0)
+            .reshape(1, -1, self.num_grid_per_side, self.num_grid_per_side)
+        )
         for t, h, w in grid_thw:
-            h_idxs = np.linspace(0, num_grid_per_side - 1, h)
-            w_idxs = np.linspace(0, num_grid_per_side - 1, w)
-
-            h_idxs_floor = h_idxs.astype(int)
-            w_idxs_floor = w_idxs.astype(int)
-            h_idxs_ceil = (h_idxs.astype(int) + 1).clip(max=num_grid_per_side - 1)
-            w_idxs_ceil = (w_idxs.astype(int) + 1).clip(max=num_grid_per_side - 1)
-
-            dh = h_idxs - h_idxs_floor
-            dw = w_idxs - w_idxs_floor
-
-            idx_list[0].extend(
-                ((h_idxs_floor * num_grid_per_side)[None].T + w_idxs_floor[None])
-                .flatten()
-                .tolist()
-                * t
-            )
-            idx_list[1].extend(
-                ((h_idxs_floor * num_grid_per_side)[None].T + w_idxs_ceil[None])
-                .flatten()
-                .tolist()
-                * t
-            )
-            idx_list[2].extend(
-                ((h_idxs_ceil * num_grid_per_side)[None].T + w_idxs_floor[None])
-                .flatten()
-                .tolist()
-                * t
+            pos_embed = torch.nn.functional.interpolate(
+                embeds, size=(h, w), mode="bilinear"
```

Comment:
I don't agree with this change. Bilinear interpolation is only used in real-time, low-quality scenarios, whereas the default bicubic interpolation is used in higher-quality but relatively slower scenarios. We can't make this trade-off here just for the sake of higher performance.

Nevertheless  I'm not completely opposed to this solution. We can make a feature option to toggle the options, leaving the end-user to make the trade-off decision.

Reply:
@terfendail , you are right, here's pos embedding, it should use bilinear. Please update to the following to behave more identical with the original implementation.
```
torch.nn.functional.interpolate(
    embeds, size=(h, w), mode="bilinear", align_corners=True
)
```
For the image resize, the default mode should be bicubic.